Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -366,6 +375,16 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, EntityUid? us
         CreateEffect(gun, ev, user);
     }
 
+    public void CauseImpulse(EntityCoordinates fromCoordinates, EntityCoordinates toCoordinates, PhysicsComponent userPhysics)
+    {
+        var fromMap = fromCoordinates.ToMap(EntityManager);
+        var toMap = toCoordinates.ToMapPos(EntityManager);
+        var mapDirection = toMap - fromMap.Position;
+        mapDirection = toMap - fromMap.Position;
```

Comment:
duplicate assignments, remove the second one

Reply:
ah right